internal/kafka/handler.go

```diff
@@ -30,6 +36,7 @@ func (this *handler) onMessage(ctx context.Context, msg *kafka.Message, cfg *con
 
 	payloadStatus := &message.PayloadStatusMessage{}
 	sanitizedPayloadStatus := &models.PayloadStatuses{}
+	cachedPayload := &cachedPayload{}
```

Comment:
`cachedPayload` will be created each time the `onMessage()` method is called.  As a result, I don't think the data will be cached.  Am I misunderstanding?

Reply:
Can you try running this method a couple of times with the same input data and printing out the contents of the `cachedPayload` variable?